packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -250,6 +260,12 @@ class DatePicker extends React.Component {
     delete customProps.onCalendarButtonClick;
     delete customProps.onInputFocus;
 
+    const attributes = Object.assign({}, customProps);
+
+    if (required) {
+      attributes['aria-required'] = 'true';
```

Comment:
would it be helpful to always have the aria-required attribute and we toggle true or false on the required prop value? @JakeLaCombe

Reply:
I think how it is should be what we need. the aria should only be set if the input is required.

Reply:
Looks like you don't need to set `aria-required` here after all since [DateInput](https://github.com/cerner/terra-framework/blob/master/packages/terra-date-picker/src/DateInput.jsx#L161-L170) uses Input and Input will add [aria-required](https://github.com/cerner/terra-core/blob/master/packages/terra-form-input/src/Input.jsx#L109-L111) automatically if `required` is set. Having said that, you should pass `required` prop down to the `required` prop in ReactDatePicker.

Reply:
@benbcai Is this what you had in mind? https://github.com/cerner/terra-framework/pull/504/commits/14324e7a538b9f6496afbbf408ae5c2f12e73378

Reply:
@gabeparra01 Yes. If you inspect the date input when you passed in the `required` prop, it should have the required attribute and `aria-label=true`.